lib/monitoring/metrics/api_exception_counter.rb

```diff
@@ -0,0 +1,29 @@
+module Monitoring
+  module Metrics
+    class ApiExceptionCounter
+      attr_reader :registry, :pubsub, :metric_name, :docstring, :labels, :sub_event_name
+
+      def setup(registry, pubsub)
+        @registry = registry
+        @pubsub = pubsub
+        @metric_name = :conjur_http_server_request_exceptions_total
+        @docstring = 'The total number of API exceptions raised by Conjur.'
+        @labels = %i[operation exception message]
+        @sub_event_name = 'conjur.request_exception'
+
+        # Create/register the metric
+        Metrics.create_metric(self, :counter)
+      end
+
+      def update(payload)
+        metric = @registry.get(@metric_name)
+        update_labels = {
+          operation: payload[:operation],
+          exception: payload[:exception],
+          message: payload[:message]
+        }
+        metric.increment(labels: update_labels)
+      end
+    end
+  end
+end
```

Comment:
What about breaking these up into using inheritance?  If we structure these metrics like this:

```ruby
module Monitoring
  module Metrics
    class BaseMetric
      attr_reader :registry, :pubsub, :metric_name, :docstring, :labels, :sub_event_name

      def setup(registry, pubsub)
        @registry = registry
        @pubsub = pubsub

        # Create/register the metric
        Metrics.create_metric(self, :counter)
      end

      def update(payload)
        metric = @registry.get(@metric_name)
        update_labels = {
          operation: payload[:operation],
          exception: payload[:exception],
          message: payload[:message]
        }
        metric.increment(labels: update_labels)
      end
    end

    class ApiExceptionCounter < BaseMetric
      def initialize
        @metric_name = :conjur_http_server_request_exceptions_total
        @docstring = 'The total number of API exceptions raised by Conjur.'
        @labels = %i[operation exception message]
        @sub_event_name = 'conjur.request_exception'
      end
    end
  end
end
```

It'll allow us to reduce the metric classes to their bare minimum definition without breaking the interface (`setup` and `update` methods).

Reply:
_Note_: Just noticed that the `update_labels` variable is unique.  As the rest of the call appears to be constant across implementations, I've updated my suggestion to include that in the child class.

Reply:
@jvanderhoof this gets a bit more complicated with the metrics that aren't simple counters. I'm not sure we'll really gain much by refactoring this once we consider that there are only 2 counter metrics and the others will still need custom logic in the `update` method.

Reply:
I agree @szh.  Let's keep these more verbose until we have a full set of potential metrics established.  If it makes sense to refactor to reduce code duplication, we can do that when we have a more complete picture of the range of metrics.